### ISSUE TITLE
New Relic Exporter Throws Panic With Zipkin Receiver

### ISSUE THREAD

**Issue by disfluxly** [open-telemetry/opentelemetry-collector-contrib#348]
**Describe the bug**
Using the opentelemetry-collector-contrib:0.4 docker image, setting up a Zipkin Receiver and a New Relic Exporter results in the New Relic Exporter throwing a Go Panic.

**Steps to reproduce**

- ./docker-compose.yml
- ./otel-collector-config.yml

**docker-compose.yml**

```
---
version: "3"
services:
  collector:
    image: otel/opentelemetry-collector-contrib:0.4.0
    command: ["--config=/etc/otel-collector-config.yml"]
    ports:
      - "9411:9411"
    environment:
      - NEW_RELIC_API_KEY=${NEW_RELIC_API_KEY}
    volumes:
      - "./otel-collector-config.yml:/etc/otel-collector-config.yml"
```

**otel-collector-config.yml**

```
receivers:
  zipkin:

exporters:
  newrelic:
    apikey: ${NEW_RELIC_API_KEY}
    timeout: 30s
  logging:
    loglevel: debug

service:
  pipelines:
    traces:
      receivers: [zipkin]
      exporters: [newrelic, logging]
```

**Sample Zipkin Data**
```
[{
    "annotations": [
        {
            "timestamp": 1592830465192113,
            "value": "test"
        }
    ],
    "duration": 601000,
    "id": "5d65e4fc46b29f3a",
    "kind": "CLIENT",
    "localEndpoint": {
        "serviceName": "testservice"
    },
    "name": "test_service",
    "tags": {
        "annotations.termination_code": "--",
        "http.request.client_ip": " 172.29.0.1",
        "http.request.method": "GET",
        "http.request.url": "http://localhost:8080/",
        "http.request.user_agent": "curl/7.54.0",
        "http.response.content_length": "70",
        "http.response.status": "200",
        "metadata.connections.active": "1",
        "metadata.connections.backend": "0",
        "metadata.connections.frontend": "1",
        "metadata.connections.retries": "0",
        "metadata.connections.server": "0",
        "metadata.connections.total_time_connect": "0",
        "metadata.queue.backend": "0",
        "metadata.queue.server": "0",
        "metadata.queue.total_time_queued": "0"
    },
    "timestamp": 1592830465192113,
    "traceId": "7739d2fc08f3213f61f39dba5d2cc626"
}]
```

**Commands**

```
export NEW_RELIC_API_KEY={Your NR Insights API Key}
docker-compose up

curl -XPOST --data-binary '[{"annotations":[{"timestamp":1592830465192113,"value":"test"}],"duration":601000,"id":"5d65e4fc46b29f3a","kind":"CLIENT","localEndpoint":{"serviceName":"testservice"},"name":"test_service","tags":{"annotations.termination_code":"--","http.request.client_ip":" 172.29.0.1","http.request.method":"GET","http.request.url":"http://localhost:8080/","http.request.user_agent":"curl/7.54.0","http.response.content_length":"70","http.response.status":"200","metadata.connections.active":"1","metadata.connections.backend":"0","metadata.connections.frontend":"1","metadata.connections.retries":"0","metadata.connections.server":"0","metadata.connections.total_time_connect":"0","metadata.queue.backend":"0","metadata.queue.server":"0","metadata.queue.total_time_queued":"0"},"timestamp":1592830465192113,"traceId":"7739d2fc08f3213f61f39dba5d2cc626"}]' http://localhost:9411/api/v2/spans
```

**What did you expect to see?**
Traces to appear in New Relic

**What did you see instead?**
Collector reporting a panic:

```
collector_1  | 2020/06/22 13:18:29 http: panic serving 172.30.0.1:40226: runtime error: invalid memory address or nil pointer dereference
collector_1  | goroutine 24 [running]:
collector_1  | net/http.(*conn).serve.func1(0xc0003fcdc0)
collector_1  |  /usr/local/go/src/net/http/server.go:1772 +0x139
collector_1  | panic(0x2628de0, 0x4615460)
collector_1  |  /usr/local/go/src/runtime/panic.go:975 +0x3e3
collector_1  | github.com/open-telemetry/opentelemetry-collector-contrib/exporter/newrelicexporter.(*transformer).SpanAttributes(0xc0004e9610, 0xc0005f1ad0, 0x46441c0)
collector_1  |  /home/circleci/project/exporter/newrelicexporter/transformer.go:94 +0x41
collector_1  | github.com/open-telemetry/opentelemetry-collector-contrib/exporter/newrelicexporter.(*transformer).Span(0xc0004e9610, 0xc0005f1ad0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
collector_1  |  /home/circleci/project/exporter/newrelicexporter/transformer.go:73 +0x3e0
collector_1  | github.com/open-telemetry/opentelemetry-collector-contrib/exporter/newrelicexporter.exporter.pushTraceData(0xc00004d240, 0xc00020af00, 0x30081a0, 0xc0004e5aa0, 0xc0005ebfc0, 0xc0004101c0, 0xc00000e320, 0x1, 0x1, 0x2adda64, ...)
collector_1  |  /home/circleci/project/exporter/newrelicexporter/newrelic.go:90 +0x128
collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.traceDataPusherOld.withObservability.func1(0x30081a0, 0xc0004e5aa0, 0xc0005ebfc0, 0xc0004101c0, 0xc00000e320, 0x1, 0x1, 0x2adda64, 0xa, 0x1, ...)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/exporter/exporterhelper/tracehelper.go:96 +0xc9
collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.(*traceExporterOld).ConsumeTraceData(0xc00000cc40, 0x30081a0, 0xc0004e4a50, 0xc0005ebfc0, 0xc0004101c0, 0xc00000e320, 0x1, 0x1, 0x2adda64, 0xa, ...)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/exporter/exporterhelper/tracehelper.go:48 +0xa7
collector_1  | go.opentelemetry.io/collector/consumer/converter.(*internalToOCTraceConverter).ConsumeTraces(0xc000496d30, 0x30081a0, 0xc0004e4a50, 0xc000373d20, 0x0, 0xc00000e310)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/consumer/converter/converter.go:43 +0x125
collector_1  | go.opentelemetry.io/collector/processor.traceFanOutConnector.ConsumeTraces(0xc00000ce60, 0x2, 0x2, 0x30081a0, 0xc0004e4a50, 0xc000373d20, 0xc000373d40, 0xc00000e2f8)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/processor/fanoutconnector.go:152 +0xad
collector_1  | go.opentelemetry.io/collector/consumer/converter.(*ocToInternalTraceConverter).ConsumeTraceData(0xc000496e00, 0x30081a0, 0xc0004e4a50, 0xc0005eba40, 0x0, 0xc00000e2f0, 0x1, 0x1, 0x2ad6f80, 0x6, ...)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/consumer/converter/converter.go:96 +0x87
collector_1  | go.opentelemetry.io/collector/receiver/zipkinreceiver.(*ZipkinReceiver).ServeHTTP(0xc0001b42a0, 0x2ffe2e0, 0xc0004120e0, 0xc0001b0d00)
collector_1  |  /home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.3.1-0.20200615181810-b28cbd9c8e2e/receiver/zipkinreceiver/trace_receiver.go:355 +0x56c
collector_1  | net/http.serverHandler.ServeHTTP(0xc0001b2700, 0x2ffe2e0, 0xc0004120e0, 0xc0001b0d00)
collector_1  |  /usr/local/go/src/net/http/server.go:2807 +0xa3
collector_1  | net/http.(*conn).serve(0xc0003fcdc0, 0x30080e0, 0xc0005eb340)
collector_1  |  /usr/local/go/src/net/http/server.go:1895 +0x86c
collector_1  | created by net/http.(*Server).Serve
collector_1  |  /usr/local/go/src/net/http/server.go:2933 +0x35c
```

Response back to curl was:

```
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9411 (#0)
> POST /api/v2/spans HTTP/1.1
> Host: localhost:9411
> User-Agent: curl/7.54.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 833
> 
* upload completely sent off: 833 out of 833 bytes
* Empty reply from server
* Connection #0 to host localhost left intact
curl: (52) Empty reply from server
```

**What version did you use?**
v0.4.0

**What config did you use?**
opentelemetry-collector-config.yml, described above

**Environment**
OS: Docker

**Additional context**
Initially thought the problem was that I didn't have 'annotations', which is an optional field according to Zipkin. I thought perhaps this was defaulting to an empty list in the transformation and thus causing a nil pointer somewhere in the New Relic code. Adding in a test annotation didn't solve. Thus I have tested with and without annotations. I have also tested with and without tags.

**Comment by jrcamp** [comment]
@MrAlias looks like you added the initial implementation so assigning to you to triage

**Comment by MrAlias** [comment]
Thanks for the awesome detail in this issue @disfluxly, sorry I haven't gotten to it sooner. I was able to reproduce the error with the steps you gave.

I also tested against the v0.8.0 release of the collector and received the same panic.

Based on the location of the panic:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/23dfc784deb8348246dca10161c1280ec1bdda97/exporter/newrelicexporter/transformer.go#L94

it seems like the exporter received a span with a `nil` status.

This seems like expected behavior based on the proto:

https://github.com/open-telemetry/opentelemetry-proto/blob/2470dbc299d297863609499a50553435094bcebe/opentelemetry/proto/trace/v1/trace.proto#L223-L225

I'm going to put in a PR to handle this correctly as a success if nil.
